Library/src/lab1/Library.java

```diff
@@ -2,10 +2,83 @@
 
 /**
  * A representation of a library of books.
- * @author <a href="mailto:rkhatchadourian@citytech.cuny.edu">Raffi Khatchadourian</a>.
+ *
+ * @author <a href="mailto:rkhatchadourian@citytech.cuny.edu">Raffi
+ * Khatchadourian</a>.
  */
 public class Library {
-    // TODO: Add the missing implementation to this class
+
+    String address;
+    Book[] collection = {};
+
+    public Library(String location) {
+        address = location;
+    }
+
+    public void addBook(Book book) {
+        Book[] add = new Book[collection.length + 1];
+        for (int i = 0; i < collection.length; i++) {
+            add[i] = collection[i];
+        }
+        add[collection.length] = book;
+        collection = add;
+    }
+
+    public void borrowBook(String title) {
+        boolean inCollection = false;
+        for (int i = 0; i < collection.length; i++) {
+            if (collection[i].getTitle().equals(title)) {
+                if (collection[i].isBorrowed()) {
+                    System.out.println("Sorry, this book is already borrowed.");
+                    inCollection = true;
+                } else {
+                    collection[i].rented();
+                    System.out.println("You successfully borrowed " + title);
+                    inCollection = true;
```

Comment:
:-1: If the library has multiple copies of the same book, you will borrow all of them.